dev/com.ibm.ws.install.map/src/com/ibm/ws/install/map/InstallMap.java

```diff
@@ -406,17 +413,49 @@ else if (jars.length == 1) {
     @SuppressWarnings({ "unchecked" })
     private URL[] getJars(File installDir) {
         JarFile installKernelJar = null;
+        JarFile repositoryResolverJar = null;
+        boolean replaceJar = false;
         List<URL> jarURLs = new ArrayList<URL>();
         try {
             installKernelJar = new JarFile(new File(installDir, (String) data.get(INSTALL_KERNEL_JAR)));
+            File repositoryResolverFile = new File((String) data.get(REPOSITORY_RESOLVER_JAR));
+            String resolverVersion = "";
+            String resolverJarName = "";
+            if (data.get(REPOSITORY_RESOLVER_JAR) != null) {
+                repositoryResolverJar = new JarFile(repositoryResolverFile);
+                Manifest resolverManifest = repositoryResolverJar.getManifest();
+                Attributes resolverAttributes = resolverManifest.getMainAttributes();
+                resolverJarName = resolverAttributes.getValue("Bundle-SymbolicName");
+                String[] resolverVersionSplit = resolverAttributes.getValue("Bundle-Version").split("\\.");
+                for (String versionSplit : resolverVersionSplit) {
+                    if (!(versionSplit.length() > 3)) {
+                        resolverVersion = resolverVersion + versionSplit;
+                    }
+                }
+            }
             Manifest manifest = installKernelJar.getManifest();
             Attributes attributes = manifest.getMainAttributes();
             String[] requireBundles = attributes.getValue("Require-Bundle").split(",");
+            String[] installMapVersionSplit = attributes.getValue("Bundle-Version").split("\\.");
+            String installMapVersion = "";
+            for (String mapVersionSplit : installMapVersionSplit) {
+                if (!(mapVersionSplit.length() > 3)) {
+                    installMapVersion = installMapVersion + mapVersionSplit;
+                }
+            }
+            if (resolverVersion.compareTo(installMapVersion) >= 0) {
```

Comment:
Currently we want to match the exact version i.e. Should not replace the jar even the version of the jar from Maven Central is higher than than the one in runtime.